packages/core/admin/admin/src/content-manager/components/BlocksEditor/index.js

```diff
@@ -232,67 +231,82 @@ const renderLeaf = ({ attributes, children, leaf }) => {
   return <span {...attributes}>{children}</span>;
 };
 
-const BlocksEditor = React.forwardRef(({ intlLabel, name, readOnly, required, error }, ref) => {
-  const { formatMessage } = useIntl();
-  const theme = useTheme();
-  const [editor] = React.useState(() => withReact(createEditor()));
-
-  const label = intlLabel.id
-    ? formatMessage(
-        { id: intlLabel.id, defaultMessage: intlLabel.defaultMessage },
-        { ...intlLabel.values }
-      )
-    : name;
-
-  /** Editable is not able to hold the ref, https://github.com/ianstormtaylor/slate/issues/4082
-   *  so with "useImperativeHandle" we can use ReactEditor methods to expose to the parent above
-   *  also not passing forwarded ref here, gives console warning.
-   */
-  React.useImperativeHandle(
-    ref,
-    () => ({
-      focus() {
-        ReactEditor.focus(editor);
-      },
-    }),
-    [editor]
-  );
-
-  return (
-    <>
-      <Flex direction="column" alignItems="stretch" gap={1}>
-        <Flex gap={1}>
-          <Typography variant="pi" fontWeight="bold" textColor="neutral800">
-            {label}
-            {required && <TypographyAsterisk textColor="danger600">*</TypographyAsterisk>}
-          </Typography>
+const BlocksEditor = React.forwardRef(
+  ({ intlLabel, name, readOnly, required, error, onChange, value }, ref) => {
+    const { formatMessage } = useIntl();
+    const theme = useTheme();
+    const [editor] = React.useState(() => withReact(createEditor()));
+
+    const label = intlLabel.id
+      ? formatMessage(
+          { id: intlLabel.id, defaultMessage: intlLabel.defaultMessage },
+          { ...intlLabel.values }
+        )
+      : name;
+
+    /** Editable is not able to hold the ref, https://github.com/ianstormtaylor/slate/issues/4082
+     *  so with "useImperativeHandle" we can use ReactEditor methods to expose to the parent above
+     *  also not passing forwarded ref here, gives console warning.
+     */
+    React.useImperativeHandle(
+      ref,
+      () => ({
+        focus() {
+          ReactEditor.focus(editor);
+        },
+      }),
+      [editor]
+    );
+
+    return (
+      <>
+        <Flex direction="column" alignItems="stretch" gap={1}>
+          <Flex gap={1}>
+            <Typography variant="pi" fontWeight="bold" textColor="neutral800">
+              {label}
+              {required && <TypographyAsterisk textColor="danger600">*</TypographyAsterisk>}
+            </Typography>
+          </Flex>
+
+          <Slate
+            editor={editor}
+            initialValue={
+              value?.blocks || [{ type: 'paragraph', children: [{ type: 'text', text: '' }] }]
+            }
+            onChange={(state) => {
+              const isAstChange = editor.operations.some((op) => op.type !== 'set_selection');
```

Comment:
I don't get what isAstChange means

Reply:
I believe it stands for `isAbstractSyntaxTreeChange`.

The check means the onChange will be fired when there is an actual change in the content, but not when you select existing content or move your cursor around existing content. Otherwise it fires on every interaction.

I was just following the docs: https://docs.slatejs.org/walkthroughs/06-saving-to-a-database, should I change the variable name or add a comment?

Reply:
I added a comment to try and clarify